src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImpl.java

```diff
@@ -619,22 +619,59 @@ private String determineKfsOwnershipTypeCodeFromPmwTaxClassificationCode(Payment
         kfsVendorDataWrapper.getVendorDetail().setVendorFirstLastNameIndicator(true);
-        kfsVendorDataWrapper.getVendorDetail().setVendorFirstName(pmwVendor.getRequestingCompanyLegalFirstName());
-        kfsVendorDataWrapper.getVendorDetail().setVendorLastName(pmwVendor.getRequestingCompanyLegalLastName());
+        kfsVendorDataWrapper.getVendorDetail().setVendorLastName(truncateLegalLastNameToMaximumAllowedLength(pmwVendor.getRequestingCompanyLegalLastName()));
+        kfsVendorDataWrapper.getVendorDetail().setVendorFirstName(truncateLegalFirstNameToMaximumAllowedLengthWhenFormattedWithLegalLastName(pmwVendor.getRequestingCompanyLegalLastName(), pmwVendor.getRequestingCompanyLegalFirstName()));
         return kfsVendorDataWrapper;
     }
 
+    private String truncateBusinessLegalNameToMaximumAllowedLength(String businessLegalName) {
+        if (businessLegalName.length() <= VendorConstants.MAX_VENDOR_NAME_LENGTH) {
+            return businessLegalName;
+        } else {
+            String truncatedBusinessLegalName = businessLegalName.substring(0, VendorConstants.MAX_VENDOR_NAME_LENGTH);
+            LOG.info("truncateBusinessLegalNameToMaximumAllowedLength: Received businessLegalName '" + businessLegalName
+                    + "' with length of " + businessLegalName.length() + " and it is being truncated to '"
+                    + truncatedBusinessLegalName + "' with length of " + truncatedBusinessLegalName.length());
+            return truncatedBusinessLegalName;
+        }
+    }
+
+    private static String truncateLegalLastNameToMaximumAllowedLength(String legalLastName) {
+        if (legalLastName.length() <= VendorConstants.MAX_VENDOR_NAME_LENGTH) {
+            return legalLastName;
+        } else {
+            String truncatedLegalLastName = legalLastName.substring(0, VendorConstants.MAX_VENDOR_NAME_LENGTH);
+            LOG.info("truncateBusinessLegalNameToMaximumAllowedLength: Received legalLastName '" + legalLastName
+                    + "' with length of " + legalLastName.length() + " and it is being truncated to '"
+                    + truncatedLegalLastName + "' with length of " + truncatedLegalLastName.length());
+            return truncatedLegalLastName;
+        }
+    }
+
+    private static String truncateLegalFirstNameToMaximumAllowedLengthWhenFormattedWithLegalLastName(String legalLastName, String legalFirstName) {
+        int maxAllowedLengthOfFirstName = VendorConstants.MAX_VENDOR_NAME_LENGTH - VendorConstants.NAME_DELIM.length() - legalLastName.length();
+        if (legalFirstName.length() <= maxAllowedLengthOfFirstName) {
+            return legalFirstName;
+        } else {
+            String truncatedLegalFirstName = legalFirstName.substring(0, maxAllowedLengthOfFirstName);
+            LOG.info("truncateLegalFirstNameToMaximumAllowedLengthWhenFormattedWithLegalLastName: Received legalFirstName '"
```

Comment:
Since the length of the last name is part of the reason how the first name is truncated, I think it may be helpful to add the last name to this log message.

Reply:
Making message more descriptive.